### ISSUE TITLE
Lambda Insights

### ISSUE THREAD

**Issue by AntonOellerer** [amazonlinux/amazon-linux-2023#734]
According to the [documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-Getting-Started-docker.html) lambda insights for container images is only available for images based off Amazon Linux 2.
It would be nice to also have this for Amazon Linux 2023

**Comment by Juneezee** [comment]
Any updates from the AWS team?

Node.js 18 has reached its end of life. We are planning to upgrade our AWS Lambda base image to Node.js 20, but the Node.js 20 base images are based on Amazon Linux 2023.

https://docs.aws.amazon.com/lambda/latest/dg/nodejs-image.html#nodejs-image-base

**Comment by alexey-tsvetnov** [comment]
Could you please clarify your question? It seems that the two previous comments together create a bit of confusion.
Are you asking to bring NodeJS 20 in AL2?

NodeJS is represented only in AL2023, that provides NodeJS 18, 20, and 22.
However, I'm not aware of AL2 supporting any NodeJS, and NodeJS 18 in particular.

```
# yum info nodejs
Loaded plugins: ovl, priorities
Error: No matching Packages to list

# cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
SUPPORT_END="2026-06-30"
```

The above link, mentioned in hte previous comment, is absolutely correct in pointing out that: `The Node.js 20 and later base images are based on the [Amazon Linux 2023 minimal container image](https://docs.aws.amazon.com/linux/al2023/ug/minimal-container.html). Earlier base images use Amazon Linux 2. AL2023 provides several advantages over Amazon Linux 2, including a smaller deployment footprint and updated versions of libraries such as glibc.`

It looks like NodeJS 18 on the AL2 image is a custom build made specifically for AWS Lambda, but the NodeJS 18 package is definitely not coming from the AL2 package repositories because it simply doesn't exist there.

**Comment by Juneezee** [comment]
> Are you asking to bring NodeJS 20 in AL2?

@alexey-tsvetnov No, you misunderstood our comments. We are asking to add Lambda Insights agent support for Amazon Linux 2023.

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-Getting-Started-docker.html

![Image](https://github.com/user-attachments/assets/ae7d7fa9-86cb-4089-a928-39f083e5f65f)

**Comment by alexey-tsvetnov** [comment]
@Juneezee , got it! Thank you for clarifying.
We'll do our best to push things forward.

**Comment by ozbenh** [comment]
So I heard back from the CloudWatch team and they said that the current agent appears to work fine on AL2023 despite what the documentation says. They are going to test further and update the doc shortly
